pkg/cloud/services/network/vpc.go

```diff
@@ -103,24 +107,45 @@ func (s *Service) reconcileVPC() error {
 		return nil
 	}
 
-	// .spec.vpc.id is nil, Create a new managed vpc.
-	if !conditions.Has(s.scope.InfraCluster(), infrav1.VpcReadyCondition) {
-		conditions.MarkFalse(s.scope.InfraCluster(), infrav1.VpcReadyCondition, infrav1.VpcCreationStartedReason, clusterv1.ConditionSeverityInfo, "")
-		if err := s.scope.PatchObject(); err != nil {
-			return errors.Wrap(err, "failed to patch conditions")
+	// .spec.vpc.id is nil. This means no managed VPC exists or we failed to save its ID before. Check if a managed VPC
+	// with the desired name exists, or if not, create a new managed VPC.
+
+	vpc, err := s.describeVPCByName()
+	if err == nil {
+		// An VPC already exists with the desired name
+
+		if !vpc.Tags.HasOwned(s.scope.Name()) {
+			return errors.Errorf("an unmanaged VPC %s already exists, refusing to create another managed VPC", vpc.ID)
 		}
```

Comment:
The VPC today is created with tags within the same call, the error message is confusing to users given that if there are no tags, there isn't another managed VPC
```suggestion
		if !vpc.Tags.HasOwned(s.scope.Name()) {
			return errors.Errorf("found VPC named %s which cannot be managed by CAPA due to lack of tags, either tag the VPC manually, or provide the `vpc.id` field instead if you wish to bring your own VPC", vpc.ID)
		}
```

Reply:
done